File: LeetCode/88. Merge Sorted Array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        x, y, z = m - 1, n - 1, len(nums1) - 1
        while y >= 0:
            if x>=0 and nums1[x] >= nums2[y]:
                nums1[z] = nums1[x]
                x -= 1
            else:
                nums1[z] = nums2[y]
                y -= 1
            z -= 1

File: LeetCode/88. Merge Sorted Array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int x = m - 1, y = n - 1, z = n + m - 1;
        while (y >= 0) {
            if (x >= 0 && (nums1[x] >= nums2[y])) {
                nums1[z] = nums1[x];
                x--;
            } else {
                nums1[z] = nums2[y];
                y--;
            }
            z--;
        }
    }
}
